File: 11. Pendalaman String/F. Gaya Penulisan Variabel/f_gaya_penulisan_variabel.pas
program f_gaya_penulisan_variabel;
var
    S : string;
    i : integer;
    isCaps : boolean;
begin
    readln(S);
    isCaps := false;
    for i:=1 to length(S) do begin
        if (isCaps = true) then 
            begin
                write(chr(ord(S[i]) xor 32));
                isCaps := false;
            end
        else if (S[i] >= 'A') and (S[i] <= 'Z') then
            write('_', chr(ord(S[i]) xor 32))
        else if (S[i] = '_') then
            isCaps := true
        else
            write(S[i]);
    end;
    writeln;
end.

File: 11. Pendalaman String/F. Gaya Penulisan Variabel/f_gaya_penulisan_variabel.py
S = input()
isCaps = False
for i in S:
    if isCaps:
        print(chr(ord(i) ^ 32), end="")
        isCaps = False
    elif i.isupper():
        print("_", chr(ord(i) ^ 32), sep="", end="")
    elif i == "_":
        isCaps = True
    else:
        print(i, end="")
print()
